feat(Storybook): Initial setup of how storybook can be used (#240)
packit/dashboard: CONTRIBUTING.md

[Technical details](https://github.com/guard/listen/wiki/Increasing-the-amount-of-inotify-watchers#the-technical-details)

## Developing frontend with Storybook

> For the dashboard's frontend, we make use of Storybook to test out different scenarios that each component can be in. For example, testing that everything looks fine for accessibility point-of-view, or that network errors displays correctly
> 
> Storybook allows us to isolate individual React components and display them in different scenarios that we can view with minimal interferance. Allowing us to do a proper integration test
> 
> However, Storybook itself is not a replacement for making sure the code works correctly. The API endpoints being used might change in the future, which is why it is good to test your PRs out against stage before production deploymnent

Reviewer: ```suggestion
For the dashboard's frontend, we make use of Storybook to test out different scenarios that each component can be in. For example, testing that everything looks fine from an accessibility point-of-view, or that network errors are displayed correctly.

Storybook allows us to isolate individual React components and display them in different scenarios that we can view with minimal interference. Allowing us to do a proper integration test.

However, Storybook itself is not a replacement for making sure the code works correctly. The API endpoints being used might change in the future, which is why it is good to test your PRs out against stage before production deployment.
```

```jsx
// Import React, it's necessary for JSX
import React from "react";
// Importing the component we want to write a story about
import { NotFound } from "./NotFound";

// Default export is the actual story page itself, it sets things up for the actual stories - which are the named exports down below
export default {
  title: "NotFound",
  component: NotFound,
};

// Here we initialize the component with default args from Storybook
// If we wanted to add some parameter ourselves that the viewer of the Story cannot change we would do so here
// Note: It's recommended to allow the user to change the variables through the Storybook website unless it's critical to the story
const Template = (args) => <NotFound {...args} />;

// Next we call bind, this allows us to use `this` properly within the component
// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/bind
export const Default = Template.bind({});
```

#### Network usage

> Most of our components need to use some sort of network activity, likely an API call to Packit service. For those stories, we are using a library called [MSJ](https://mswjs.io/).

Reviewer: ```suggestion
Most of our components need to use some sort of network activity, likely an API call to Packit service. For those stories, we are using a library called [MSW](https://mswjs.io/).
```

> Most of our componments need to use some sort of network activity, likely an API call to Packit service. For those stories, we are using a library called [MSJ](https://mswjs.io/).

Reviewer: ```suggestion
Most of our components need to use some sort of network activity, likely an API call to Packit service. For those stories, we are using a library called [MSJ](https://mswjs.io/).
```

Author: Fixed, thanks
